server/mailet/integration-testing/src/main/java/org/apache/james/mailets/TemporaryJamesServer.java

```diff
@@ -155,7 +157,10 @@ private TemporaryJamesServer(File workingDir, MailetContainer mailetContainer, S
             .overrideWith((binder) -> binder.bind(PersistenceAdapter.class).to(MemoryPersistenceAdapter.class))
             .overrideWith(additionalModules)
             .overrideWith(new TestJMAPServerModule())
-            .overrideWith((binder) -> binder.bind(WebAdminConfiguration.class).toInstance(WebAdminConfiguration.TEST_CONFIGURATION));
+            .overrideWith((binder) -> {
+                binder.bind(WebAdminConfiguration.class).toInstance(WebAdminConfiguration.TEST_CONFIGURATION);
+                binder.bind(UploadRepository.class).to(UnsupportedUploadRepository.class);
+            });
```

Comment:
There is something wrong here.

`UnsupportedUploadRepository` is an heresie.

Please instead provide a modular enough design to only expose operation supported by the james server that is running.

Likely some dark magic with a multibinder should do the trick.

Reply:
fixed